src/users/infrastructure/users.prisma.repository.ts

```diff
@@ -0,0 +1,15 @@
+import { UsersRepository } from '../domain/users.repository';
+import { prisma } from '../../database/database';
+import { User } from '../domain/users.entity';
+
+export const projectPrismaRepository: UsersRepository = {
+  async getUsers(): Promise<User[]> {
+    try {
+      const users = await prisma.user.findMany();
```

Comment:
Hello @Cabrakana!

In that case you get all the users and this is not the goal of getUser because you should only get the data of one user (via email). Please check that and apply this change.

```
async getUser(email: string): Promise<User | null> {
    try {
      const user = await prisma.user.findUnique({
        where: {
          email: email,
        },
      });
      return user;
    } catch (error) {
      console.error('Error fetching user:', error);
      throw new Error('Could not fetch user');
    }
  },
};
```